pkg/storaged/utils.js

```diff
@@ -1140,3 +1140,28 @@ export function get_mount_points(client, block_fsys, subvol) {
 
     return mounted_at;
 }
+
+export function get_byte_units(guide_value) {
+    const byte_suffixes = [null, "KB", "MB", "GB", "TB", "PB", "EB", "ZB"];
+
+    function unit(index) {
+        return {
+            name: byte_suffixes[index],
+            factor: Math.pow(1000, index)
+        };
+    }
+
+    const units = [unit(2), unit(3), unit(4)];
+
+    // The default unit is the largest one that gives us at least
+    // two decimal digits in front of the comma.
+
+    for (let i = units.length - 1; i >= 0; i--) {
```

Comment:
... and do a `units.find((unit, i) => { ...}` here, for functional l33tness.

This isn't a review blocker, just in case you feel like it.

Reply:
If I did anything to this code I was actually planning to integrate this check into the `unit()` function directly and drop the loop...

...but ah, I just noticed the `break;`.  Boo.

Reply:
Yeah, `.find()` gives you that semantics.

Reply:
I ended up using `Math.max(0, findLastIndex)` which gives me the "...find the last-matching thing, or 0 otherwise" semantics.